File: erp-frontend/src/enums/biz/contractStatus.ts
import { BaseEnum, BaseEnumItem } from '@/enums/baseEnum';

/**
 * 合同状态枚举
 */
const CONTRACT_STATUS: BaseEnum<number, string> = new BaseEnum<number, string>();
CONTRACT_STATUS.set('WAIT_CREATE', new BaseEnumItem<number, string>(0, '待生成合同任务'));
CONTRACT_STATUS.set('TASK_EXECUTING', new BaseEnumItem<number, string>(1, '任务执行中'));
CONTRACT_STATUS.set('TASK_RETURN', new BaseEnumItem<number, string>(3, '任务退修'));
CONTRACT_STATUS.set('CONTRACT_CLOSE', new BaseEnumItem<number, string>(2, '合同关闭'));
CONTRACT_STATUS.set('TASK_CLOSE', new BaseEnumItem<number, string>(4, '任务关闭'));
CONTRACT_STATUS.set('WAIT_DELIVERY', new BaseEnumItem<number, string>(5, '待交付'));

export { CONTRACT_STATUS };

File: erp-backend/xingyun-shkb/src/main/java/com/lframework/xingyun/shkb/enums/ContractStatus.java
package com.lframework.xingyun.shkb.enums;

import com.baomidou.mybatisplus.annotation.EnumValue;
import com.lframework.starter.web.core.enums.BaseEnum;

/**
 * 合同状态
 */

public enum ContractStatus implements BaseEnum<Integer> {
    WAIT_CREATE(0, "待生成合同任务"),
    TASK_EXECUTING(1, "任务执行中"),
    CONTRACT_CLOSE(2, "合同关闭"),
    TASK_RETURN(3, "任务退修"),
    TASK_CLOSE(4, "任务关闭"),
    WAIT_DELIVERY(5, "待交付")
    ;
    @EnumValue
    private final Integer code;

    private final String desc;

    ContractStatus(Integer code, String desc) {

        this.code = code;
        this.desc = desc;
    }

    @Override
    public Integer getCode() {

        return this.code;
    }

    @Override
    public String getDesc() {

        return this.desc;
    }
}
